### PR TITLE
Add vSphere extension to testgrid conformance tests

### PR DESCRIPTION
**Pull request by briantopping**
**How to categorize this PR?**

/area dev-productivity
/kind test
/platform vsphere

**What this PR does / why we need it**:
Add vSphere extension to Kubernetes testgrid via Google Cloud VMware Engine

**Special notes for your reviewer**:
Needs review WRT test machinery and testgrid integrations. Submitted as draft, also see XXX for testgrid triggers.

**Release note**:

```other developer
Add vSphere extension to testgrid conformance tests.
```

### CODE REVIEW

**Comment by briantopping** [comment]
Review questions:
* Need to check the sequencing of the build for the docker image that TM runs. The docker image adds OpenVPN, which is required to access the Google Cloud VMware instance. Where can I find the default makefile targets that are executed in CI? I'll link the image build to the appropriate target.
* There’s a bunch of configuration for the SERVICEACCOUNT_JSON that comes from the GCP build. I’ve created that as a Secret called `shoot-operator-vsphere`, how do I propagate that out to staging?
* When ready to test in staging, how is it updated? That project doesn't allow forks.
* Need review of the difference between `.ci/test-integration.sh` and `.test-defs/gcve-infrastructure-test.yaml`. I see that both are required but lost the differences between them. Does documentation cover this anywhere?

**Comment by dguendisch** [comment]
I'm probably missing the big picture: what do you actually try to achieve? I'm not familiar with https://github.com/gardener/gardener-extension-provider-vsphere/blob/main/test/integration/infrastructure/infrastructure_test.go but as far as I can see it is not related to conformance tests in any way?
Regular conformance tests are already performed on our staging landscape, hence if you wanted to include vsphere, you'd have to adapt the test configuration there.

cc @MartinWeindel @hendrikKahl

**Comment by briantopping** [comment]
@dguendisch This is the project I was discussing with you at the beginning of the month. The summary is getting the vSphere provider integrated into the staging landscape, but automation to create and tear down Google Cloud VMware instances for the target environment. Other work in the mean time was learning GCVE constraints, their docs are not released yet.

**Comment by dguendisch** [comment]
Here are a few pointers:
- https://github.com/gardener/gardener-extension-provider-vsphere/blob/main/test/integration/infrastructure/infrastructure_test.go seems to do some vsphere specific tests, I see you amended this test in this PR but as said, this has nothing to do with testgrid or conformance, it's unrelated.
- this infrastructure_test.go was probably used by @MartinWeindel to locally test the extension (I guess without testmachinery support, as the old vsphere installation could not interface network wise with github.com...)
- if you want to include vsphere as a cloudprovider into the testgrid/conformance tests that we already periodically execute for all Gardener supported cloud providers, you'd need to amend that in our internal staging landscape, the aformentioned infrastructure_test.go is some test that is not related / not used when launching the "official" conformance tests

**Review by tuxgoose — COMMENTED** [review]
@briantopping  is this ready for review?

Oops, looks like you marked it ready for review 👍

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,7 @@
 EXTENSION_PREFIX            := gardener-extension
 NAME                        := provider-vsphere
 VALIDATOR_NAME              := validator-vsphere
+GCVE_TM_RUN_NAME    		:= gcve-tm-run
```
Comment (tuxgoose) [inline Makefile:18]: @briantopping  is this indent correct?

Comment (briantopping) [reply Makefile]: Fixed, thanks! Had incorrect makefile settings in my IDE

**Review by tuxgoose — APPROVED** [review]
LGTM

**Pull request merged**